integrationtests/run_tests.bash

```diff
@@ -79,17 +126,42 @@ else
 	echo skipping clean step
 fi
 
+
+
 echo "=====> compile ($SRC_DIR)"
 mkdir -p $ROS_HOME
 mkdir -p $CATKIN_DIR/src
 mkdir -p $TEST_RESULT_TARGET_DIR
 if ! [ -d $SRC_DIR ]
 then
 	ln -s $ORIG_SRC $SRC_DIR
-	ln -s $ORIG_SRC/Tools/sitl_gazebo ${CATKIN_DIR}/src/mavlink_sitl_gazebo
+	# Symbolic links to catkin packages below. Note that submodules like
+	# rotors_simulator and mav_comm contain many catkin packages.
+	# Symbolic link for the rotors_simulation sub-module
+	#ln -s $ORIG_SRC/Tools/sitl_gazebo ${CATKIN_DIR}/src/mavlink_sitl_gazebo
+	ln -s $ORIG_SRC/Tools/rotors_simulator/rotors_gazebo ${CATKIN_DIR}/src/rotors_gazebo
+	ln -s $ORIG_SRC/Tools/rotors_simulator/rotors_gazebo_plugins ${CATKIN_DIR}/src/rotors_gazebo_plugins
+	ln -s $ORIG_SRC/Tools/mav_comm/mav_msgs ${CATKIN_DIR}/src/mav_msgs
+	
 fi
 cd $CATKIN_DIR
-catkin_make
+
+# These build parameters are used by the CMakeLists.txt in the
+# rotors_gazebo_plugins and rotors_gazebo packages.
+# ADDITIONAL_INCLUDE_DIRS=...				This assumes mav_msgs has been symlinked to the catkin workspace as per above
+# BUILD_MAVLINK_INTERFACE_PLUGIN=TRUE		Build the MAVLink interface plugin for Gazebo
+# BUILD_OCTOMAP_PLUGIN=FALSE				Do not build the Octomap plugin for Gazebo
+# BUILD_OPTICAL_FLOW_PLUGIN=TRUE			Build the optical flow plugin for Gazebo
+# NO_ROS=TRUE 								Instruct the Gazebo plugins to build without ROS dependencies. Note that even though ROS is used for these tests, PX4 uses the Gazebo plugins without any ROS dependencies.
+# CMAKE_MODULE_PATH=$OPTICAL_FLOW_INSTALL_DIR		Tell catkin where to find the OpticalFlow installation, so find(OpticalFlow) works.
```

Comment:
@gbmhunter  @devbharat @LorenzMeier I think we need to discuss how to do this cleanly.  What you have done here breaks one of my major use cases.